File: Program29.java
import java.lang.*;
import java.util.*;

// Time Complexity : O(N)

class Factors
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Entered value is:"+this.iNo);
    }

    public void DisplayFactors()
    {
        int iCnt = 0;
    
        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
          if((iNo % iCnt) == 0)
          {
             System.out.println(iCnt);
          }
        }
    }
}

class Program29
{
    public static void main(String arg[])
    {
       Factors fobj = new Factors();

       fobj.Accept();
       fobj.Display();

       fobj.DisplayFactors();
    }
}

File: Program29.cpp
#include<iostream>
using namespace std;

// Time Complexity : O(N)

void DisplayFactors(int iNo)
{
    int iCnt = 0;

    for(iCnt = 1; iCnt <= iNo; iCnt++)
    {
      if((iNo % iCnt) == 0)
      {
         cout<<iCnt<<endl;
      }
    }
}

int  main()
{
    int iValue = 0;
    
    cout<<"Enter number"<<endl;
    cin>>iValue;

    DisplayFactors(iValue);

    return 0;
}